src/utils/promptRunner.ts

```diff
@@ -0,0 +1,107 @@
+import { spawn } from "child_process";
+import { window } from "vscode";
+const output = window.createOutputChannel("Docker Labs: AI Tools");
+
+export const getRunArgs = (promptRef: string, projectDir: string, username: string, platform: string, pat: string, render = false) => {
+    const isLocal = promptRef.startsWith('local://');
+    let promptArgs: string[] = ["--prompts", promptRef];
+    let mountArgs: string[] = [];
+
+    if (isLocal) {
+        const localPromptPath = promptRef.replace('local://', '');
+        const pathSeparator = platform === 'win32' ? '\\' : '/';
+        promptRef = localPromptPath.split(pathSeparator).pop() || 'unknown-local-prompt';
+        promptArgs = ["--prompts-dir", `/app/${promptRef}`];
+        mountArgs = ["--mount", `type=bind,source=${localPromptPath},target=/app/${promptRef}`];
+    }
+
+    const baseArgs: string[] = [
+        'run',
+        '--rm',
+        '-v', '/var/run/docker.sock:/var/run/docker.sock',
+        '-v', 'openai_key:/root',
+        '--mount', 'type=volume,source=docker-prompts,target=/prompts'
+    ];
+
+    const runArgs: string[] = render ? [] : [
+        'vonwig/prompts:latest',
+        ...(render ? [] : ['run']),
+        "--host-dir", projectDir,
+        "--user", username,
+        "--platform", platform,
+        ...promptArgs,
+        '--jsonrpc',
+        '--pat', pat
```

Comment:
This doesn't seem right to me.  Will pat be nil sometimes?

Reply:
Correct. If the prompt image can't handle nil then I can make sure to skip it.